__tests__/services/users.service.spec.ts

```diff
@@ -1,32 +1,87 @@
 import { Container } from 'typedi';
 import { genSaltSync, hashSync } from 'bcrypt';
 import { UsersService } from '@services/users.service';
+import { factory } from 'typeorm-seeding';
+import { User } from '@entities/user.entity';
+import { getRepository, Repository } from 'typeorm';
+import { mockUpdateResult } from '../utils/mocks';
+import { HashInvalidError } from '@exception/users/hashinvalid.error';
+import { HashExpiredError } from '@exception/users/hashexpired.error';
 
 let usersService: UsersService;
+let user: User;
+let userRepository: Repository<User>;
 
-beforeAll(async () => {
-  usersService = Container.get(UsersService);
-});
-
-describe('compare password', () => {
-  let userPassword: string;
+describe('UsersService', () => {
+  beforeAll(async () => {
+    usersService = Container.get(UsersService);
+  });
 
-  beforeEach(() => {
-    userPassword = hashSync('password', genSaltSync());
+  it('all dependencies should be defined', () => {
+    expect(usersService).toBeDefined();
   });
+  describe('comparePassword', () => {
+    let userPassword: string;
+
+    beforeEach(() => {
+      userPassword = hashSync('password', genSaltSync());
+    });
 
-  it('checks that the password matches', () => {
-    const hashedPassword = 'password';
-    const result = usersService.comparePassword({
-      password: hashedPassword,
-      userPassword
+    it('checks that the password matches', () => {
+      const hashedPassword = 'password';
+      const result = usersService.comparePassword({
+        password: hashedPassword,
+        userPassword
+      });
+      expect(result).toBeTruthy();
+    });
+
+    it("checks that the password don't match", () => {
+      const password = 'anotherpassword';
+      const result = usersService.comparePassword({ password, userPassword });
+      expect(result).toBeFalsy();
     });
-    expect(result).toBeTruthy();
   });
+  describe('verifyUser', () => {
+    beforeEach(async () => {
+      user = await factory(User)().create();
+    });
+
+    beforeAll( () => {
+      usersService = Container.get(UsersService);
+      userRepository = getRepository<User>(User);
+    });
 
-  it("checks that the password don't match", () => {
-    const password = 'anotherpassword';
-    const result = usersService.comparePassword({ password, userPassword });
-    expect(result).toBeFalsy();
+    it('should throw error if the hash is invalid', async () => {
+      user.verifyHash = '11111111-2222-3333-4444-555555555555';
```

Comment:
take in mind that you could use faker to avoid hard-coding the mocks just as a suggestion.